hivemq-edge/src/frontend/src/extensions/datahub/components/nodes/OperationNode.tsx

```diff
@@ -1,34 +1,44 @@
-import { FC } from 'react'
+import { FC, useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
-import { Handle, NodeProps, Position } from 'reactflow'
+import { NodeProps, Position } from 'reactflow'
 import { HStack, Text, VStack } from '@chakra-ui/react'
 
-import { DataHubNodeType, OperationData } from '../../types.ts'
-import { styleSourceHandle } from '../../utils/node.utils.ts'
+import { DataHubNodeType, FunctionDefinition, OperationData } from '../../types.ts'
 import NodeIcon from '../helpers/NodeIcon.tsx'
 import { NodeWrapper } from './NodeWrapper.tsx'
+import { CustomHandle } from '@/extensions/datahub/components/nodes/CustomHandle.tsx'
 
 export const OperationNode: FC<NodeProps<OperationData>> = (props) => {
   const { t } = useTranslation('datahub')
   const { data, id, type } = props
   const { action } = data
 
+  const model = useMemo(() => {
+    if (!action) return undefined
+
+    return typeof action === 'string'
+      ? ({
+          functionId: action,
+        } as FunctionDefinition)
+      : action
+  }, [action])
```

Comment:
suggestion: fully valid, but a bit hard to parse. WDYT about this one?

```suggestion
  const model = useMemo(() => {
    if (!action) return undefined
    
    if (typeof action === 'string') {
      return {
        functionId: action,
      } satisfies FunctionDefinition
    }

    return action
  }, [action])
```